src/api/mutations/amenity/useSaveAmenitiesMutation.ts

```diff
@@ -0,0 +1,37 @@
+import { useMutation } from '@tanstack/react-query';
+import toast from 'react-hot-toast';
+import { Amenities, EditAmenitiesResponse } from '../../../types/amenity.types';
+import httpClient from '../../httpClient';
+
+type Params = {
+  amenitiesToSave: Amenities;
+  isNewAccomodation: boolean;
+};
+
+const createAmenities = async (amenitiesToSave: Amenities, accomodationId: string) => {
+  const { data } = await httpClient.post<EditAmenitiesResponse>(
+    `/amenities/${accomodationId}`,
+    amenitiesToSave
+  );
+  return data;
+};
+
+const updateAmenities = async (amenitiesToSave: Amenities, accomodationId: string) => {
+  const { data } = await httpClient.put<EditAmenitiesResponse>(
+    `/amenities/${accomodationId}`,
```

Comment:
create constants for route names so you can reuse them, 
e.g. 
// routes.ts

`export const AMENITIES_ROUTE = '/amenities';`